test/tests.ts

```diff
@@ -0,0 +1,179 @@
+import { TASK_COMPILE_GET_SOURCE_PATHS } from "@nomiclabs/buidler/builtin-tasks/task-names";
+import { BuidlerContext } from "@nomiclabs/buidler/internal/context";
+import { assert } from "chai";
+import { readFileSync } from "fs";
+import { join } from "path";
+
+export async function expectErrorAsync(
+  f: () => Promise<any>,
+  errorMessage?: string
+) {
+  try {
+    await f();
+  } catch (err) {
+    assert.equal(err.message, errorMessage);
+  }
+}
+
+async function unloadBuidlerContext() {
+  if (BuidlerContext.isCreated()) {
+    const ctx = BuidlerContext.getBuidlerContext();
+    const globalAsAny = global as any;
+    if (ctx.environment !== undefined) {
+      for (const key of Object.keys(ctx.environment)) {
+        globalAsAny[key] = undefined;
+      }
+      // unload config file too.
+      unloadModule(ctx.environment.config.paths.configFile);
+    }
+    BuidlerContext.deleteBuidlerContext();
+  }
+
+  // Unload all the buidler's entrypoints.
+  unloadModule("@nomiclabs/buidler");
+  unloadModule("@nomiclabs/buidler/config");
+  unloadModule("@nomiclabs/buidler/register");
+  unloadModule("@nomiclabs/buidler/internal/cli/cli");
+  unloadModule("@nomiclabs/buidler/internal/lib/buidler-lib");
+  unloadModule("@nomiclabs/buidler/internal/core/config/config-env");
+  unloadModule("@nomiclabs/buidler/internal/core/tasks/builtin-tasks");
+
+  // Unload bultin tasks.
+  Object.keys(require.cache).forEach(module => {
+    if (module.includes("buidler/builtin-tasks")) {
+      unloadModule(module);
+    }
+  });
+}
+
+function unloadModule(path: string) {
+  try {
+    delete require.cache[require.resolve(path)];
+  } catch (err) {
+    // module wasn't loaded
+  }
+}
+
+async function reset() {
+  await unloadBuidlerContext();
+  unloadModule("../src/index");
+  return require("@nomiclabs/buidler");
+}
+
+describe("Solpp plugin", async function() {
+  describe("js-config-project", async function() {
+    before("setup", async function() {
+      process.chdir(__dirname + "/js-config-project");
+    });
+
+    beforeEach("reset buidler context", async function() {
+      this.env = await reset();
+    });
+
+    it("should evaluate symbols as javascript functions", async function() {
+      const paths = await this.env.run(TASK_COMPILE_GET_SOURCE_PATHS);
+      const generatedContractA = readFileSync(paths[0]).toString();
+      assert.include(generatedContractA, "1337");
+    });
+
+    it("should compile without errors", async function() {
+      try {
+        await this.env.run("compile");
+      } catch (err) {
+        assert.isUndefined(err);
```

Comment:
This is somwhat weird, you can use this: https://www.chaijs.com/api/assert/#method_doesnotthrow